### ISSUE TITLE
Error in `dynamicmap.js`: `this._markerClick` is not a function

### ISSUE THREAD

**Issue by MoritzLost** [doublesecretagency/craft-googlemaps#49]
I'm trying to initialize an info window in JavaScript, but it's not working correctly. In 4.1.1, the following code worked fine (simplified to relevant parts):

```js
const marker = {
    id: `marker-lat-lng-123,456`,
    lat: 123,
    lng: 456,
};
const infoWindowHtml = '<p>the title</p>';
map.markers(marker, {
    infoWindowOptions: {
        content: infoWindowHtml,
    },
})
```

While this doesn't cause any errors, the info window doesn't open when the marker is clicked. I tried to attach an event listener to open it manually using the `markerClick` option, at which point I get the following error message in the console:

> Uncaught (in promise) TypeError: this._markerClick is not a function
> markers dynamicmap.js:174

Indeed it looks like the `_markerClick` function [doesn't exist anymore in `dynamicmap.js`](https://github.com/doublesecretagency/craft-googlemaps/blob/v4-dev/src/resources/js/dynamicmap.js). I checked the JS file in 4.1.1, which still [includes the _markerClick function](https://github.com/doublesecretagency/craft-googlemaps/blob/4.1.1/src/resources/js/dynamicmap.js#L755). Regression bug?

In addition to this bug, what is the 'correct' way to attach simple info windows to markers in 4.1.5?

- PHP 8.1.3
- Craft 3.7.37
- Google Maps 4.1.5

**Comment by lindseydiloreto** [comment]
Definitely a regression bug. Good catch, thanks for reporting it!

I'll follow up with you via Discord DM to resolve this and #48.

**Comment by MoritzLost** [comment]
@lindseydiloreto Thanks! I've tried the latest branch and it does fix the error message – but I still can't get the info window to show up. I've tried to show all info windows just to test if they're being created at all, but it looks like they aren't:

```js
const marker = {
    id: `marker-lat-lng-123,456`,
    lat: 123,
    lng: 456,
};
const infoWindowHtml = '<p>the title</p>';
map.markers(marker, {
    infoWindowOptions: {
        content: infoWindowHtml,
    },
    markerClick: () => map.openInfoWindow('*'),
})
```

This doesn't do anything. I've also put some logging inside `dynamicmap.js` just to check, looks like `this._infoWindows` is completely empty, even after calling the `map.markers` with the `infoWindowOptions` shown above. I've looked through the 4.1.1 branch again, in this version, the `map.markers` function [included some code to attach the info window](https://github.com/doublesecretagency/craft-googlemaps/blob/4.1.1/src/resources/js/dynamicmap.js#L163-L166). Those lines are gone in 4.1.5, so maybe the info windows aren't being attached at all?
